docs/src/selectors.md

```diff
@@ -284,6 +284,61 @@ converts `'//html/body'` to `'xpath=//html/body'`.
 
 Attribute engines are selecting based on the corresponding attribute value. For example: `data-test-id=foo` is equivalent to `css=[data-test-id="foo"]`, and `id:light=foo` is equivalent to `css:light=[id="foo"]`.
 
+## Selecting element from the list
+
+Sometimes page contains a number of indistinguishable elements. For example:
+
+```html
+<ul>
+  <li>Item<button>Buy</button></li>
+  <li>Item<button>Buy</button></li>
+  <li>Item<button>Buy</button></li>
+</ul>
+```
+
+Playwright supports `:nth-match(inner selector, index)` to target a particular item in this case. It will query the inner selector and return an element with the specified `index` from the query result. Note that index is one-based.
```

Comment:
It is unclear what `inner selector` is in this context, let's steer this from being a spec to being a guide and start with the example that you have below.